uimaj-core/src/main/java/org/apache/uima/resource/PearSpecifier.java

```diff
@@ -41,18 +43,57 @@
   public void setPearPath(String aPearPath);
 
   /**
-   * Gets parameters that may be read by the pear resource class when it is initialized.
+   * Gets legacy string-valued parameters that may be read by the pear resource class when it is
+   * initialized. These parameters are represented as follows in the PEAR specifier XML:
+   * 
+   * <pre>{@code
+   * <parameters>
+   *   <parameter name="param1" value="val1"/>
+   * </parameters>  
+   * }</pre>
    * 
-   * @return an array of parameters.  This will never return <code>null</code>.
+   * @return an array of parameters. This will never return <code>null</code>.
+   * 
+   * @deprecated These parameters only support string values. Better use
+   *             {@link #setPearParameters(NameValuePair[])}.
    */
   public Parameter[] getParameters();
 
   /**
-   * Sets parameters that may be read by the pear resource class when it is initialized.
+   * Sets legacy string-valued parameters that may be read by the pear resource class when it is 
+   * initialized.
    * 
    * @param parameters the Parameters to set.
+   * 
+   * @see #getParameters()
+   * @deprecated These parameters only support string values. Better use
+   *             {@link #getPearParameters()}.
```

Comment:
should link to setPearParameters, not get ?

Reply:
Indeed. Fixed.